src/wasm/syscalls_openmp.cpp

```diff
@@ -263,54 +273,58 @@ namespace wasm {
      * @param    chunk     The chunk size for the parallel loop
      *
      * The functions compute the upper and lower bounds and strides to be used for the
-     * set of iterations to be executed by the current thread from the statically
-     * scheduled loop that is described by the initial values of the bounds, strides,
-     * increment and chunks for parallel loop and distribute constructs.
+     * set of iterations to be executed by the current thread.
+     *
+     * The guts of the implementation in openmp can be found in __kmp_for_static_init in
+     * runtime/src/kmp_sched.cpp
      *
      * See sched_type for supported scheduling.
      */
     WAVM_DEFINE_INTRINSIC_FUNCTION(env, "__kmpc_for_static_init_4", void, __kmpc_for_static_init_4,
-                                   I32 loc, I32 gtid, I32 schedule, I32 _plastiter, I32 _plower,
-                                   I32 _pupper, I32 _pstride, I32 incr, I32 chunk) {
+                                   I32 loc, I32 gtid, I32 schedule, I32 lastIterPtr, I32 lowerPtr,
+                                   I32 upperPtr, I32 stridePtr, I32 incr, I32 chunk) {
         const std::shared_ptr<spdlog::logger> &logger = util::getLogger();
         logger->debug("S - __kmpc_for_static_init_4 {} {} {} {} {} {} {} {} {}",
-                      loc, gtid, schedule, _plastiter, _plower, _pupper, _pstride, incr, chunk);
+                      loc, gtid, schedule, lastIterPtr, lowerPtr, upperPtr, stridePtr, incr, chunk);
 
-        // Assumes the pointers will be valid throughout the function
-        Runtime::Memory *memoryPtr = getExecutingModule()->defaultMemory;
-        I32 *plower = &Runtime::memoryRef<I32>(memoryPtr, _plower);
-        I32 *pupper = &Runtime::memoryRef<I32>(memoryPtr, _pupper);
-        I32 *pstride = &Runtime::memoryRef<I32>(memoryPtr, _pstride);
-        I32 *plastiter = &Runtime::memoryRef<I32>(memoryPtr, _plastiter);
-//        logger->debug("Thread {}: lower {}, upper {}, lastiter {}, stride {}", thisThreadNumber, *plower, *pupper,
-//                      *plastiter, *pstride);
+        // Get host pointers for the things we need to write
+        I32 *lastIter = &ompMemoryRef<I32>(lastIterPtr);
+        I32 *lower = &ompMemoryRef<I32>(lowerPtr);
+        I32 *upper = &ompMemoryRef<I32>(upperPtr);
+        I32 *stride = &ompMemoryRef<I32>(stridePtr);
+
+        printf("Points %i: lower %p, upper %p, lastiter %p, stride %p\n", thisThreadNumber, lower, upper,
+                      lastIter, stride);
+
+        logger->debug("Thread {}: lower {}, upper {}, lastiter {}, stride {}", thisThreadNumber, *lower, *upper,
+                      *lastIter, *stride);
 
         if (sectionThreadCount == 1) {
-            *plastiter = true;
-            *pstride = (incr > 0) ? (*pupper - *plower + 1) : (-(*plower - *pupper + 1));
-            return;
+            throw std::runtime_error("Not yet supported running loop with one thread");
         }
```

Comment:
What evidence do you have of this? it works fine with me?

Reply:
None actually, I was just trying to reduce the amount of code I needed to look at earlier. You can put that back now